pyensign/ensign.py

```diff
@@ -100,6 +101,19 @@ async def publish(self, topic_name, *events, on_ack=None, on_nack=None):
         if topic_name == "":
             raise ValueError("topic_name is required")
 
+        if not isinstance(topic_name, str):
+            raise TypeError(
+                "expected type 'str' for topic_name, got '{}'".format(
+                    type(topic_name).__name__
+                )
+            )
+
+        # Handle a list of events provided as a single argument
+        if len(events) == 1 and (
+            isinstance(events[0], list) or isinstance(events[0], tuple)
+        ):
+            events = events[0]
```

Comment:
Needed to add this to support use cases like:

```python
await ensign.publish(topic, [event_1, event_2])
await ensign.publish(topic, (event_1, event_2))
```

Otherwise, the list or tuple gets interpreted as a single argument, but we want to parse it as multiple events.

Reply:
this is a great feature that adds flexibility for users!